From 5025171aae5a79a883117a8f2fa727da44883a32 Mon Sep 17 00:00:00 2001
From: Alona Solntseva <alona.solntseva@tandemg.com>
Date: Tue, 31 May 2022 13:50:51 +0300
Subject: [PATCH] [SW-0] tpm: pcr extend code cleanup

Clean and organize tpm_hash_and_pcr_extend code
Remove unneeded changes

Signed-off-by: Alona Solntseva <alona.solntseva@tandemg.com>
Change-Id: If6a260c6ad9fbd8510679bedd93c8dd3ad9d21b9
---

diff --git a/arch/gaudi2/common/tpm_auth.c b/arch/gaudi2/common/tpm_auth.c
index 81eab9e..b258ef4 100644
--- a/arch/gaudi2/common/tpm_auth.c
+++ b/arch/gaudi2/common/tpm_auth.c
@@ -106,18 +106,11 @@
 
 int tpm_hash_and_pcr_extend(uint32_t image_index, uintptr_t image_ptr, uint32_t image_size)
 {
-	int ret;
 	uint8_t hash[SHA256_SUM_LEN] = { 0 };
 
 	sha256_csum((void *)image_ptr, image_size, hash);
 
-	ret = tpm2_pcr_extend(image_index, TPM2_ALG_SHA256, hash, SHA256_SUM_LEN);
-	if (ret) {
-		preboot_puts("preboot_pcr_extend: pcr extend failed\n");
-		cpu_boot_err_set(CPU_BOOT_ERR_TPM_FAIL);
-	}
-
-	return ret;
+	return tpm_pcr_extend(image_index, hash);
 }
 
 void preboot_tpm_auth(void)
